src/Lib/CrashDetector.php

```diff
@@ -0,0 +1,108 @@
+<?php declare(strict_types=1);
+
+/*
+ Copyright (c) 2023, Manticore Software LTD (https://manticoresearch.com)
+
+ This program is free software; you can redistribute it and/or modify
+ it under the terms of the GNU General Public License version 2 or any later
+ version. You should have received a copy of the GPL license along with this
+ program; if you did not, you can find it at http://www.gnu.org/
+ */
+
+namespace Manticoresearch\Buddy\Lib;
+
+use Manticoresearch\Buddy\Network\ManticoreClient\Settings;
+
+class CrashDetector {
+	/**
+	 * Initialize crash detector by using manticore settings
+	 * @param Settings $settings
+	 * @return void
+	 */
+	public function __construct(protected Settings $settings) {
+	}
+
+	/**
+	 * The method check searchd.log and try to find if we crashed last time
+	 * and send metric that increments crash counter
+	 * @return bool
+	 * 	we return false even when we cannot detect that we had crash
+	 */
+	public function hadCrash(): bool {
+		// If we have no log set, skip it
+		if (!$this->settings->searchdLog) {
+			return false;
+		}
+
+		// Let's parse searchd.log now and find out last crash
+		$fp = fopen($this->settings->searchdLog, 'rb');
+		if (!$fp) {
+			return false;
+		}
+		$hadCrash = static::hadCrashInLog($fp);
+		fclose($fp);
+
+		return $hadCrash;
+	}
+
+	/**
+	 * Get resource to the file and read it from the end to try to find crash log
+	 * @param mixed $fp
+	 * @return bool
+	 */
+	protected static function hadCrashInLog(mixed $fp): bool {
+		/** @var resource $fp */
+		$hadCrash = false;
+		fseek($fp, 0, SEEK_END);
+		$pos = ftell($fp) - 1;
```

Comment:
Reading by one char seems to be suboptimal. Why don't you read by 4KB (seems to be a typical sector size)? In theory reading 1 byte or 4KB  should give about the same latency, but to read 4KB by 1 byte may take 4096 times more time. Of course you can rely on different caches (storage cache, page cache), but still it seems to make more sense to read 4KB right away.